### PR TITLE
muffon: init at 2.0.3

### PR DESCRIPTION
**Pull request by octodi**
## Description of changes

Added muffon a [Advanced multi-source music streaming client](https://muffon.netlify.app/)

https://github.com/staniel359/muffon

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/mu/muffon/package.nix
Comment (LordGrimmauld) [inline pkgs/by-name/mu/muffon/package.nix:40]: Avoid using `with lib;` as the version reference will escape the with binding.
Further, just `agpl3` isn't a thing anymore, either agpl3Only or agpl3Plus, muffon looks like agpl3Only to me.
```suggestion
  meta = {
    description = "Advanced multi-source music streaming client";
    homepage = "https://muffon.netlify.app/";
    changelog = "https://github.com/staniel359/muffon/releases/tag/v${version}";
    license = lib.licenses.agpl3Only;
    maintainers = with lib.maintainers; [ octodi ];
    mainProgram = "muffon";
    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
    platforms = [ "x86_64-linux" ];
  };
```

File: pkgs/by-name/mu/muffon/package.nix
Comment (LordGrimmauld) [inline pkgs/by-name/mu/muffon/package.nix:20]: Muffon 2.0.3 will install it with the version string stripped away already.
```suggestion

```

File: pkgs/by-name/mu/muffon/package.nix
Comment (LordGrimmauld) [inline pkgs/by-name/mu/muffon/package.nix:9]: muffon recently updated to 2.0.3:
```suggestion
  version = "2.0.3";
```

File: pkgs/by-name/mu/muffon/package.nix
Comment (LordGrimmauld) [inline pkgs/by-name/mu/muffon/package.nix:12]:
```suggestion
    hash = "sha256-2eLe/xvdWcOcUSE0D+pMOcOYCfFVEyKO13LiaJiZgX0=";
```

File: pkgs/by-name/mu/muffon/package.nix
Comment (LordGrimmauld) [inline pkgs/by-name/mu/muffon/package.nix:26]: `--replace` is deprecated. Consider using `--replace-fail` or `--replace-warn`
```suggestion
      --replace-fail 'Exec=AppRun' 'Exec=${pname}'
```

**Review by LordGrimmauld — CHANGES_REQUESTED** [review]

**Comment by LordGrimmauld** [comment]
You'll also want to rebase onto a more up-to-date master branch, or else these changes won't really reflect what would be the result of merging.

**Review by LordGrimmauld — APPROVED** [review]
Alright! Built and tested on `linux-x86_64`.
Both nixpkgs-review and nixpkgs-hammering still runs fine, it is indeed pulling v2.0.3, sound and search works. LGTM.

File: pkgs/by-name/mu/muffon/package.nix
Comment (drupol) [inline pkgs/by-name/mu/muffon/package.nix:11]: Don't re-use `pname`, use a string literal please.

File: pkgs/by-name/mu/muffon/package.nix
Comment (drupol) [inline pkgs/by-name/mu/muffon/package.nix:1]: Pass your file through `nixfmt-rfc-style`

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/by-name/mu/muffon/package.nix
Comment (octodi) [reply pkgs/by-name/mu/muffon/package.nix:11]: Like this? 
```nix
{
  lib,
  fetchurl,
  appimageTools,
  makeWrapper,
}:

let
  version = "2.0.3";
  src = fetchurl {
    url = "https://github.com/staniel359/muffon/releases/download/v${version}/muffon-${version}-linux-x86_64.AppImage";
    hash = "sha256-2eLe/xvdWcOcUSE0D+pMOcOYCfFVEyKO13LiaJiZgX0=";
  };
  appimageContents = appimageTools.extractType2 { pname = "muffon"; inherit src version; };
in
appimageTools.wrapType2 {
  pname = "muffon";
  inherit src version;

  extraInstallCommands = ''
    source "${makeWrapper}/nix-support/setup-hook"
    wrapProgram $out/bin/muffon \
      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
    install -m 444 -D ${appimageContents}/muffon.desktop -t $out/share/applications
    substituteInPlace $out/share/applications/muffon.desktop \
      --replace-fail 'Exec=AppRun' 'Exec=muffon'
    install -m 444 -D ${appimageContents}/muffon.png \
      $out/share/icons/hicolor/512x512/apps/muffon.png
  '';

  meta = {
    description = "Advanced multi-source music streaming client";
    homepage = "https://muffon.netlify.app/";
    changelog = "https://github.com/staniel359/muffon/releases/tag/v${version}";
    license = lib.licenses.agpl3Only;
    maintainers = with lib.maintainers; [ octodi ];
    mainProgram = "muffon";
    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
    platforms = [ "x86_64-linux" ];
  };
}```

Comment (LordGrimmauld) [reply pkgs/by-name/mu/muffon/package.nix:11]: Yes, true. I forgot src should stay independent from package name, that'll work

Comment (drupol) [reply pkgs/by-name/mu/muffon/package.nix:11]: If you really want me to check what you have changed in the file, please just copy the changed files only, it will be easier and faster for me to check.

Comment (LordGrimmauld) [reply pkgs/by-name/mu/muffon/package.nix:11]: Actually is there any good/reliable resources/documentation on where to reuse pname and where to type it out?

I see muffon/pname used in:
- src (where explicit is recommended)
- extraInstallCommands
- meta (changelog link and such)

With sources not using pname makes sense, and drupols comment only referred to sources as far as i can tell, but i do still wonder about the other two.

Comment (octodi) [reply pkgs/by-name/mu/muffon/package.nix]: @LordGrimmauld Could you please suggest changes in the new commit

**Comment by LordGrimmauld** [comment]
You'll have to squash those commits. Changes look fine, but then again they looked fine to my untrained eye even before drupol found the pname thing.

**Comment by LordGrimmauld** [comment]
CI checks don't work, seems you forgot the other commit

**Comment by octodi** [comment]
> CI checks don't work, seems you forgot the other commit

Now that was a codespace issue, it was add trailing newline in the end

**Review by LordGrimmauld — APPROVED** [review]
Aight, lgtm still. But then again, there might be issues i am unqualified to find, so still waiting for @drupol

**Comment by drupol** [comment]
I'm waiting for the CI to be done before merging anything... I'm going to check tomorrow morning, this browser tab will stay open until there.

**Review by drupol — APPROVED** [review]

**Pull request merged**
